lib/grape/roar/representer.rb

```diff
@@ -9,7 +9,11 @@ def self.included(base)
 
       module ClassMethods
         def represent(object, _options = {})
-          object.extend self
+          object.extend(self) unless object.singleton_class == object.class
+        rescue TypeError => e
+          # What do we do here?
```

Comment:
I personally think that we should fail gracefully here since obtaining `nil` or passing the wrong data around in a duck-typed language like Ruby is fairly commonplace. Thoughts?

Reply:
Obtaining `nil` could be commonplace, but is handled above and wouldn't raise an exception. I think I'd want to fail loudly when attempting to `present` an incompatible type.

Reply:
I agree with @joeyAghion, makes sense to me to fail loudly in error cases.

Reply:
Works for me! Thanks for the input 👍

Reply:
+1 to either just letting `TypeError` through and documenting it or re-raising a specific exception from this lib